addon/components/polaris-text-field.js

```diff
@@ -504,21 +515,88 @@ export default Component.extend({
     return inputType === 'number' && !disabled;
   }).readOnly(),
 
-  checkFocus() {
-    let { input, focused } = this.getProperties('input', 'focused');
+  normalizedValue: computed('value', function() {
+    let value = this.get('value');
+    return value !== null ? value : '';
+  }),
 
-    if (input && focused) {
-      // TODO this _seems_ to work in Polaris without the row below, but not for us (it does not apply focus class)
-      // look what's here....though this seems to do the job for now
-      this.set('focus', true);
```

Comment:
This was removed in favor of the `focus` computed property below now that the react implementation has updated their [code](https://github.com/Shopify/polaris-react/blob/bba36af2ad10ba9626bba168b52e2e9222ac4c60/src/components/TextField/TextField.tsx#L149-L157) to check the previous and current value of `focused` in order to determine whether or not to focus or blur the field.

We can't really do the same thing and call `blur` on `didUpdateAttrs` because there's a chance _other_ attributes have been updated while `focused` is set to `false`, and we wouldn't want to accidentally blur the field as the user is typing just because the initial value of `focused` was `false`.

Reply:
>We can't really do the same thing and call `blur` on `didUpdateAttrs` because there's a chance _other_ attributes have been updated while `focused` is set to `false`, and we wouldn't want to accidentally blur the field as the user is typing just because the initial value of `focused` was `false`.

Not sure I follow this 😬 Can't we track `wasFocused` internally and call `blur`/`focus` as appropriate from `didUpdateAttrs`? If `focused` changes, then we should still be obeying it, and if you're worried about "accidentally blurring the field as the user is typing", that's on the host app to pass sensible property values 🤷‍♂️

Reply:
I guess we can make some bootleg version of `wasFocused`. The original buggyness I was seeing wasn't a host app being irresponsible, it was that you couldn't type without the field blur-ing unless you explicitly passed `focused=true`, which shouldn't happen. I'll see about making our own `wasFocused`.

Reply:
Yeah, I don't see why this shouldn't work with `wasFocused` (we're doing similar things in some [other components](https://github.com/smile-io/ember-polaris/blob/update/polaris-v3.10.0/addon/components/polaris-resource-list.js#L800) already)